llvm/lib/Target/AMDGPU/AMDGPUPromoteAlloca.cpp

```diff
@@ -459,6 +422,36 @@ static Value *GEPToVectorIndex(GetElementPtrInst *GEP, AllocaInst *Alloca,
   return ConstantInt::get(GEP->getContext(), Quot);
 }
 
+// Function to check if a Value is an operand of a GetElementPtrInst.
+static bool isValueInGEP(GetElementPtrInst *GEP, Value *ValueToCheck) {
```

Comment:
This still feels like too specific of a check, and the same form of pattern could arise from other instructions. It would be better to detect it as part of the transformation